packages/db/src/search/RuleSearch.ts

```diff
@@ -0,0 +1,109 @@
+  */
+  protected buildBasicQuery(knex: Knex): {
+    countQuery: Knex.QueryBuilder,
+    searchQuery: Knex.QueryBuilder,
+  } {
+    const {
+      collections: collectionsTable,
+      providers: providersTable,
+    } = TableNames;
+
+    const countQuery = knex(this.tableName)
+      .count(`${this.tableName}.cumulus_id`);
+
+    const searchQuery = knex(this.tableName)
+      .select(`${this.tableName}.*`)
+      .select({
+        collectionName: `${collectionsTable}.name`,
+        collectionVersion: `${collectionsTable}.version`,
+        provider: `${providersTable}.name`,
+      });
+
+    if (this.searchCollection()) {
+      searchQuery.innerJoin(collectionsTable, `${this.tableName}.collection_cumulus_id`, `${collectionsTable}.cumulus_id`);
+      countQuery.innerJoin(collectionsTable, `${this.tableName}.collection_cumulus_id`, `${collectionsTable}.cumulus_id`);
+    } else {
+      searchQuery.leftJoin(collectionsTable, `${this.tableName}.collection_cumulus_id`, `${collectionsTable}.cumulus_id`);
+    }
+
+    if (this.searchProvider()) {
+      searchQuery.innerJoin(providersTable, `${this.tableName}.provider_cumulus_id`, `${providersTable}.cumulus_id`);
+      countQuery.innerJoin(providersTable, `${this.tableName}.provider_cumulus_id`, `${providersTable}.cumulus_id`);
+    } else {
+      searchQuery.leftJoin(providersTable, `${this.tableName}.provider_cumulus_id`, `${providersTable}.cumulus_id`);
+    }
+
+    return { countQuery, searchQuery };
+  }
+
+  /**
+  * Build queries for infix and prefix
+  *
+  * @param params
+  * @param params.countQuery - query builder for getting count
+  * @param params.searchQuery - query builder for search
+  * @param [params.dbQueryParameters] - db query parameters
+  */
+  protected buildInfixPrefixQuery(params: {
+    countQuery: Knex.QueryBuilder,
+    searchQuery: Knex.QueryBuilder,
+    dbQueryParameters?: DbQueryParameters,
+  }) {
+    const { countQuery, searchQuery, dbQueryParameters } = params;
+    const { infix, prefix } = dbQueryParameters ?? this.dbQueryParameters;
+    if (infix) {
+      [countQuery, searchQuery].forEach((query) => query.whereLike(`${this.tableName}.name`, `%${infix}%`));
+    }
+    if (prefix) {
+      [countQuery, searchQuery].forEach((query) => query.whereLike(`${this.tableName}.name`, `${prefix}%`));
+    }
+  }
+
+  /**
+  * Translate postgres records to api records
+  *
+  * @param pgRecords - postgres Rule records returned from query
+  * @param knex - knex for the translation method
+  * @returns translated api records
+  */
+  protected async translatePostgresRecordsToApiRecords(
+    pgRecords: PostgresRuleRecord[],
+    knex: Knex
+  ): Promise<Partial<RuleRecord>[]> {
+    log.debug(`translatePostgresRecordsToApiRecords number of records ${pgRecords.length} `);
+
+    const apiRecords = pgRecords.map(async (record) => {
+      const apiRecord = await translatePostgresRuleToApiRule(record, knex);
```

Comment:
That might have been true that if the initial query returned undefined, this might've tried again. Good catch.

In any case, we're not doing the branch logic now 😄  https://github.com/nasa/cumulus/pull/3760/commits/fd8f5bfdefec61549adc5ded6c396fcf5e3dbc8c